libtsuba/src/RDG.cpp

```diff
@@ -622,16 +639,20 @@ tsuba::RDG::SetTopologyFile(const katana::Uri& new_top) {
   return core_->RegisterTopologyFile(new_top.BaseName());
 }
 
-tsuba::RDG::RDG(std::unique_ptr<RDGCore>&& core) : core_(std::move(core)) {
+void
+tsuba::RDG::InitArrowVectors() {
   // Create an empty array, accessed by Distribution during loading
+  host_to_owned_global_ids_ = katana::EmptyChunkedArray(arrow::uint64(), 0);
```

Comment:
I'm not sure how Distribution handles these empty arrays. And I'm surprised this is working. But it doesn't block this PR from being merged.

Reply:
In `LoadPartMetadata`, Distribution.cpp calls `Unmarshal*` routines passing in `graph_pg_.local_to_user_id()` and the like.  `ArrowInterchange.h:87` calls `source->length()` and crashes.  This path is only exercised if you load a partitioned graph.  I don't think any of our CI tests load a partitioned graph.

Reply:
I got that. I meant after that, if the arrays have 0 length, I don't understand how it works.

Reply:
Oh, they just have to be initialized because the code assumes it.  The loading process creates new arrays, causing the initial arrays to become garbage.

Reply:
But in the previously partitioned graphs (for backward compatibility), it would not load anything for those arrays. That's what I'm confused about. Is that not a concern here?

Reply:
Oh, that is your department :)  I don't understand the semantics of these arrays well enough to know if I can live without them.  If not, just tell me that your check in was a breaking change and I'll repartition my graphs.

Reply:
Hehe. @scober how does `DistGraphView` handle this?

Reply:
I'm not sure I understand the question. `DistGraphView` defers almost all work to the underlying `Distribution`. So a malformed `Distribution` will result in bad/undefined behavior from the `DistGraphView`. Does that address your question?

Reply:
Yes, that's what I was expecting. Which is why I'm surprised that it works for hosts > 1. @witchel Is the Partitioner not called on these loaded graphs? That is, is it going through UpdateGraphView directly?

Reply:
So far as I know, the partitioner is not called when we load a partitioned graph.

Reply:
I can confirm that I'm getting Partition errors whenever I run updates on graphs partitioned before Roshan's recent checkin.  There is a debug assert that fires on Partition.cpp:1026 in the debug build, I think it seg faults in the release build.  I'm repartitioning all my graphs.

Reply:
Is there a convenient place to bail out with a helpful error message if we are trying to partition an outdated RDG? Or do we want to just assert/seg fault and tell people they have to update their RDGs?

Reply:
Excellent question.  I'm out of juice for this issue in this PR.  What I think is that old partitioned graphs should be destroyed, otherwise some things might work, but others might not.  Maybe only updates are broken, in which case no one but me cares, but I don't know.  This is a consequence of Roshan's #940, so far as I understand, which I would document as a breaking change for partitioned graphs.

For the sake of a single summary, I also see messy exits (misleading error message or segv) when loading a graph partitioned for N hosts on m<N and m>N hosts.  This PR does not affect that.

Reply:
Thanks, Emmet.